pilot/pkg/serviceregistry/kube/controller/serviceexportcache.go

```diff
@@ -191,8 +199,39 @@ func (ec *serviceExportCacheImpl) ExportedServices() []exportedService {
 	return out
 }
 
+func (ec *serviceExportCacheImpl) Run(stop <-chan struct{}) {
+	select {
+	case <-ec.serviceExportCh:
+	case <-stop:
+		return
+	}
+	dInformer := ec.client.DynamicInformer().ForResource(mcs.ServiceExportGVR)
+	_ = dInformer.Informer().SetTransform(kube.StripUnusedFields)
+	if ec.opts.DiscoveryNamespacesFilter != nil {
+		ec.filteredInformer = informer.NewFilteredSharedIndexInformer(ec.opts.DiscoveryNamespacesFilter.Filter, dInformer.Informer())
+	} else {
+		ec.filteredInformer = informer.NewFilteredSharedIndexInformer(nil, dInformer.Informer())
+	}
+	// Register callbacks for events.
+	ec.registerHandlers(ec.filteredInformer, "ServiceExports", ec.onServiceExportEvent, nil)
+	go ec.filteredInformer.Run(stop)
+	kube.WaitForCacheSync(stop, ec.filteredInformer.HasSynced)
+	ec.started.Store(true)
+}
+
 func (ec *serviceExportCacheImpl) HasSynced() bool {
-	return ec.filteredInformer.HasSynced()
+	return true
```

Comment:
This seems broken? its not actually synced?

If its not, can you comment in the code why its safe, because its very counter intuitive.

Reply:
This was never addressed... @hzxuzhonghu